src/lib/hooks/swap/useSwapInfo.tsx

```diff
@@ -45,8 +45,8 @@ function useComputeSwapInfo(): SwapInfo {
   )
   const trade = useBestTrade(
     isExactIn ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT,
-    parsedAmount,
-    (isExactIn ? currencyOut : currencyIn) ?? undefined
+    currencyIn && currencyOut && parsedAmount,
+    (currencyIn && currencyOut && (isExactIn ? currencyOut : currencyIn)) ?? undefined
   )
```

Comment:
could this result in a weird behavior where i.e. `currencyIn && currencyOut && parsedAmount` resolves to the value at `currencyOut` when `parsedAmount` is falsy?

Reply:
yes!